### PR TITLE
Fixing cluster tests

### PR DESCRIPTION
**Pull request by AvoidMe**

### CODE REVIEW

**Comment by AvoidMe** [comment]
Tested here: https://github.com/AvoidMe/kuberlogic/actions/runs/3084185979

File: .github/workflows/build-test-deploy.yaml [hunk 1]
```diff
@@ -29,15 +29,36 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v2
+        if: ${{ !github.event.issue.pull_request }}
         with:
           # 0 - all history, it uses in generating part of BUILD in version
           fetch-depth: 0
 
+      - uses: actions/github-script@v3
```
Comment (telepenin) [inline .github/workflows/build-test-deploy.yaml:37]: Why is it needed? The problem not relevant to the PR.

File: .github/workflows/build-test-deploy.yaml [hunk 2]
```diff
@@ -29,15 +29,36 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v2
+        if: ${{ !github.event.issue.pull_request }}
```
Comment (telepenin) [inline .github/workflows/build-test-deploy.yaml:32]: Why that needed?

File: .github/workflows/build-test-deploy.yaml [hunk 3]
```diff
@@ -53,10 +74,18 @@ jobs:
 
       - name: Cache CLI
         uses: actions/cache@v2
+        if: ${{ !github.event.issue.pull_request }}
         with:
           path: ./modules/dynamic-apiserver/bin
           key: cli-${{ github.sha }}
 
+      - name: Cache CLI from PR
```
Comment (telepenin) [inline .github/workflows/build-test-deploy.yaml:82]: Why need 2 cache CLI?

File: modules/dynamic-operator/controllers/kuberlogicservice_controller_test.go [hunk 4]
```diff
@@ -41,7 +42,7 @@ var _ = Describe("KuberlogicService controller", func() {
 	timeout := time.Second * 60
 	// in a real world backup / restore might take a little long to complete
 	if useExistingCluster() {
-		timeout = time.Second * 300
+		timeout = time.Second * 600
```
Comment (telepenin) [inline modules/dynamic-operator/controllers/kuberlogicservice_controller_test.go:45]: Please comment why timeout increased

File: modules/dynamic-apiserver/pkg/cli/install_operation.go [hunk 5]
```diff
@@ -293,7 +293,7 @@ func runInstall(k8sclient kubernetes.Interface) func(command *cobra.Command, arg
 			return err
 		}
 		command.Println("Waiting for Kuberlogic to be ready...")
-		cmd = fmt.Sprintf("%s -n kuberlogic wait --for=condition=ready pods --all --timeout=300s", kubectlBin)
+		cmd = fmt.Sprintf("%s -n kuberlogic wait --for=condition=ready pods --all --timeout=600s", kubectlBin)
```
Comment (telepenin) [inline modules/dynamic-apiserver/pkg/cli/install_operation.go:296]: Please comment why timeout increased

**Review by telepenin — COMMENTED** [review]
As I see you add some logic to the PR tests, but the tests fails on master and PR conditions inactivated.

File: modules/dynamic-apiserver/pkg/cli/install_operation.go (hunk 5 above)
Comment (AvoidMe) [reply modules/dynamic-apiserver/pkg/cli/install_operation.go]: Sometimes 300 is not enough on this tests:
https://github.com/AvoidMe/kuberlogic/actions/runs/3085186362/jobs/4988330890

File: modules/dynamic-operator/controllers/kuberlogicservice_controller_test.go (hunk 4 above)
Comment (AvoidMe) [reply modules/dynamic-operator/controllers/kuberlogicservice_controller_test.go:45]: not relevant anymore

File: .github/workflows/build-test-deploy.yaml (hunk 3 above)
Comment (AvoidMe) [reply .github/workflows/build-test-deploy.yaml]: not relevant anymore

File: .github/workflows/build-test-deploy.yaml (hunk 2 above)
Comment (AvoidMe) [reply .github/workflows/build-test-deploy.yaml]: not relevant anymore

File: .github/workflows/build-test-deploy.yaml (hunk 1 above)
Comment (AvoidMe) [reply .github/workflows/build-test-deploy.yaml]: not relevant anymore

**Review by telepenin — APPROVED** [review]
As I see you add some logic to the PR tests, but the tests fails on master and PR conditions inactivated.

**Pull request merged**